### DISCUSSION TITLE
Contact management with IOTA addresses

### DISCUSSION THREAD

**Discussion by johanneskeim** [iotaledger/firefly#1462] [Features / Enhancements]
The contacts feature should make it easy to save IOTA wallet addresses as named contacts so it's both faster to send IOTA to your friends and other personal wallets, as well as to quickly and easily identify recipients and senders in your transaction history.

It should be designed with three principles in mind: privacy, portability, and decentralization

**Comment by maxwellmattryan** [comment]
From @rafaeldjpbrochado (#497):

## Description

With the introduction of reusable addresses, contact management is now possible. I'd like to suggest giving users the ability to store & label contacts. One of the several ways in which this could be added is as a `Contacts` tab underneath the existing `Wallet` tab. Below is a quick mock-up of it when selected:

![contactstab](https://user-images.githubusercontent.com/39466455/111369121-11048c00-866d-11eb-945d-312e0c9b2ece.png)

Upon accessing the `Contacts` tab, users would see a list of their saved contacts, perhaps akin to a phone's contact list. They'd be able to `add` new contacts, `edit` and `delete` existing contacts.

In the back-end, I suppose contacts would simply be a table that stores `addresses` and `aliases` inside of Stronghold and associated with a profile, though I'm not familiar with potential technical limitations.

Alias | Address
--- | ---
Jane | atoi1qzmsfx259ughcl8yqm9t94up9ewp28zgtcmkd0kwtgne0gkyvaz6jg7ume1
John | atoi1qzmsfx259um9t94uphcl8yq0kwtgn9ewp28zgtcmkdge0gkyvaz6jg7ume2
Mom | atoi1qzmsm9t94up9ewp28zgtcmkdghcl8yq0kwtgne0gkyvafx259uz6jg7ume3
Dad | atoi1qzmsfx259um9t28zgtcmkdghcl8yq0kwtgne94up9ewp0gkyvaz6jg7ume4
Community Project | atoi1qzmsfx25tcmkdghcl8y9um9t94up9ewp28zgq0kwtgne0gkyvaz6jg7ume5

Then, when sending funds from their `Wallet`, users should have the ability to select a contact from their contact list, as opposed to always having to input a receiving address.

## Motivation

Users shouldn't have to constantly ask for their friends' addresses (in the same way that we don't constantly ask for other people's phone numbers - we just ask once and then save them). Contact management is an essential feature.

## Are you planning to do it yourself in a pull request?

No. I've been meaning to produce mock-ups for what this could look like in the app but I've been busy with other work in my day-to-day life. If someone else could pick this up, that'd be great. Otherwise I'll try to come back to this later.
